webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -175,7 +176,7 @@ export class MonacoService implements OnDestroy {
     this.editorControl.selectFileHandler(fileNode);
     if (fileNode.temp) {
       //blank new file
-      this.setMonacoModel(fileNode, <{ contents: string, etag: string, language: string }>{ contents: '', etag: '', language: '' }, true).subscribe(() => {
+      this.setMonacoModel(fileNode, <{ contents: string, etag: string, language: string }>{ contents: fileNode.model.contents, etag: '', language: '' }, true).subscribe(() => {
```

Comment:
If this is a new file, why do we need `fileNode `? Is this a bugfix for something

Reply:
If it is a new file and is not saved, then without this change the content of the file will be lost when saving the file via prompt that appears when we try to close the modified but not saved new file.

Reply:
Fixed it.